File: frontend/src/lib/types/bills.ts
import type { Party } from './party'

export type BillStatus = 'Active' | 'Committee' | 'Stalled' | 'Passed' | 'Failed'

export interface BillSummary {
  id: string
  number: string
  title: string
  sponsor: string
  party: Party
  status: BillStatus
  topics: string[]
  lastAction: string
  lastActionTimestamp: number
  summary: string
}

export interface BillSponsor {
  name: string
  bioguideId: string
  party: string
  state: string
  district: string | null
}

export interface BillCosponsor {
  name: string
  bioguideId: string
  party: string
  state: string
}

export interface BillAction {
  date: string
  text: string
  type: string
}

export interface PartyBreakdown {
  democrat:    { yea: number; nay: number }
  republican:  { yea: number; nay: number }
  independent: { yea: number; nay: number }
}

export interface BillVoteMemberPosition {
  bioguideId: string
  name:       string
  party:      string
  state:      string
  photoUrl:   string | null
  position:   string
}

export interface BillVote {
  id:              string | null
  date:            string
  chamber:         'House' | 'Senate'
  question:        string | null
  result:          string | null
  required:        string | null
  yeas:            number | null
  nays:            number | null
  present:         number | null
  notVoting:       number | null
  partyBreakdown:  PartyBreakdown | null
  memberPositions: BillVoteMemberPosition[]
  sourceUrl:       string | null
}

export interface BillDetail {
  id: string
  number: string
  title: string
  congress: number
  introducedDate: string
  status: BillStatus
  summary: string
  sponsor: BillSponsor | null
  cosponsors: BillCosponsor[]
  policyArea: string | null
  topics: string[]
  subjects: string[]
  congressGovUrl: string
  actions: BillAction[]
  votes: BillVote[]
  _hasDetailedVotes: boolean
}

File: apps/api/app/schemas/bills.py
# apps/api/app/schemas/bills.py
from pydantic import BaseModel

class BillSummary(BaseModel):
    id: str
    number: str | None = None
    title: str
    sponsor: str | None = None
    party: str | None = None
    status: str | None = None
    topics: list[str] = []
    lastAction: str | None = None
    summary: str | None = None

class SponsorDetail(BaseModel):
    name: str | None = None
    bioguideId: str | None = None
    party: str | None = None
    state: str | None = None
    district: str | None = None

class PartyBreakdown(BaseModel):
    yea: int = 0
    nay: int = 0

class VoteDetail(BaseModel):
    id: str
    date: str | None = None
    chamber: str
    question: str | None = None
    result: str
    yeas: int = 0
    nays: int = 0
    present: int = 0
    notVoting: int = 0
    partyBreakdown: dict[str, PartyBreakdown] = {}
    sourceUrl: str | None = None

class BillsByTopicResponse(BaseModel):
    slug: str
    bills: list[BillSummary]
    count: int
